sim/battle.ts

```diff
@@ -1465,6 +1465,7 @@ export class Battle {
 					sideStaleness = sideStaleness === 'external' ? sideStaleness : pokemon.staleness;
 				}
 				pokemon.activeTurns++;
+				pokemon.hasBeenActive = true;
```

Comment:
This is the wrong place to set a previouslyActive flag...

Reply:
well where would it be then?

Reply:
The point of switch-out seems like a sane place.

Reply:
Although depending on how the OM is designed, the point of move use might also work? Depends on the use-case.

Reply:
The variable name should reflect which one it tracks, though.

Reply:
`previouslySwitchedIn` vs `previouslyUsedMove`

Reply:
Should this not be a field on `m: PokemonModData;`?

Reply:
It's sufficiently general-purpose that I'm fine with it here.

Reply:
It could be, but it seems like something that could have actual use in the future